webapp/src/pages/Settings.tsx

```diff
@@ -637,90 +651,188 @@ const Settings: React.FC = () => {
-      <Paper
-        variant="outlined"
-        sx={{
-          flex: 1,
-          padding: 2,
-          overflow: "auto",
-          [`& .${formControlLabelClasses.labelPlacementStart}`]: {
-            justifyContent: "flex-end",
-            marginLeft: 0,
-          },
-          [`& .${formGroupClasses.root}`]: { marginX: 2, marginBottom: 2 },
-          [`& .fixedWidthInput .${inputClasses.root}`]: { width: "12ch" },
-          [`& .${inputClasses.input}`]: { fontSize: 14, padding: 0 },
-          [`& .${inputLabelClasses.root}`]: { fontWeight: "bold" },
-        }}
-      >
-        <Typography variant="subtitle1" marginBottom={3}>
-          View and edit certain fields from your config file. Once your changes
-          are saved, expect some delays for recomputing the affected tasks.
-        </Typography>
-        <AccordionLayout
-          name="Project Configuration"
-          description="View the fields that define the dataset to load in Azimuth."
-          link="reference/configuration/project/"
-        >
-          {getProjectConfigSection()}
-        </AccordionLayout>
-        <AccordionLayout
-          name="Model Contract Configuration"
-          description="View and edit some fields that define the ML pipelines and the metrics."
-          link="reference/configuration/model_contract/"
-        >
-          {getModelContractConfigSection()}
-        </AccordionLayout>
-        <AccordionLayout
-          name="Analyses Customization"
-          description="Enable or disable some analyses and edit corresponding thresholds."
-          link="reference/configuration/analyses/"
-        >
-          {getAnalysesCustomizationSection()}
-        </AccordionLayout>
-      </Paper>
-      <Box display="flex" justifyContent="space-between" paddingY={2}>
-        <Button
-          variant="contained"
-          onClick={() => {
-            setPartialConfig({});
-            setLanguage(undefined);
+    <DialogLayout
+      open={open}
+      onClose={onCloseDialog}
+      sx={{
+        "& .MuiDialogContent-root": {
+          padding: (theme) => theme.spacing(1),
+        },
+        "& .MuiDialogActions-root": {
+          display: "flex",
+          justifyContent: "space-between",
+          paddingX: (theme) => theme.spacing(2),
+        },
+      }}
+      titleSection={
+        <Box display="flex" justifyContent="space-between">
+          <Typography variant="subtitle1">
+            View and edit certain fields from your config file. Once your
+            changes are saved, expect some delays for recomputing the affected
+            tasks.
+          </Typography>
+          <Tooltip title="close" placement="bottom">
+            <IconButton
+              size="small"
+              color="primary"
+              onClick={() =>
+                Object.keys(partialConfig).length > 0 || isUpdatingConfig
+                  ? showModal(true)
+                  : onCloseDialog(false)
```

Comment:
Instead of a second modal, I think the simplest and most standard way to prompt for confirmation like that is to use `window.confirm()`, with something like this:
```suggestion
                (Object.keys(partialConfig).length === 0 ||
                  window.confirm(DISCARD_CONFIRM_MESSAGE)) &&
                onCloseDialog(false)
```

Reply:
True. "window. confirm" has a basic design. But we can consider using it. Thanks